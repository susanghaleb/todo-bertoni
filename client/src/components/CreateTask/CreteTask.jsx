import React, { memo, useState } from "react";
import { makeStyles } from "@material-ui/core/styles";
import Input from "@material-ui/core/Input";
import Button from "@material-ui/core/Button";
import Grid from "@material-ui/core/Grid";
import PropTypes from "prop-types";

const useStyles = makeStyles(theme => ({
  root: {
    marginTop: 100,
    display: "flex",
    justifyContent: "space-between"
  },
  button: {
    margin: theme.spacing(1)
  },
  input: {
    width: "50%",
    marginBottom: 20
  }
}));

const CreteTask = ({ submitTask }) => {
  const classes = useStyles();
  const [text, setText] = useState("");
  const handleChange = ({ target }) => setText(target.value);
  const handleClick = () => {
    submitTask(text);
    setText("");
  };

  return (
    <Grid item className={classes.root} lg={12}>
      <Input
        name="createtodo"
        value={text}
        onChange={handleChange}
        placeholder="What needs to be done?"
        required
        className={classes.input}
      />
      <Button
        classkey="send"
        variant="contained"
        color="primary"
        className={classes.button}
        onClick={handleClick}
        disabled={!text}
      >
        Add todo
      </Button>
    </Grid>
  );
};

CreteTask.propTypes = {
  submitTask: PropTypes.func.isRequired
};

export default memo(CreteTask);
