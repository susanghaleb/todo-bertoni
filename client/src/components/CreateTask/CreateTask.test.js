import React from "react";
import renderer from "react-test-renderer";
import CreateTask from "./CreteTask";

it("renders correctly", () => {
  const mockFunction = jest.fn();
  const tree = renderer
    .create(<CreateTask submitTask={mockFunction} />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
