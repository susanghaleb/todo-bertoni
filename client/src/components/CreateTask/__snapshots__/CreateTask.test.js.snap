// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="MuiGrid-root makeStyles-root-1 MuiGrid-item MuiGrid-grid-lg-12"
>
  <div
    className="MuiInputBase-root MuiInput-root MuiInput-underline makeStyles-input-3"
    onClick={[Function]}
    required={true}
  >
    <input
      className="MuiInputBase-input MuiInput-input"
      onBlur={[Function]}
      onChange={[Function]}
      onFocus={[Function]}
      placeholder="What needs to be done?"
      required={true}
      type="text"
      value=""
    />
  </div>
  <button
    className="MuiButtonBase-root MuiButton-root makeStyles-button-2 MuiButton-containedPrimary MuiButton-contained Mui-disabled Mui-disabled"
    disabled={true}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    tabIndex={-1}
    type="button"
  >
    <span
      className="MuiButton-label"
    >
      Add todo
    </span>
  </button>
</div>
`;
